apps/condo/domains/ticket/utils/serverSchema/TicketDocumentGeneration/paidWorks.js

```diff
@@ -0,0 +1,220 @@
+}
+
+const _changeToBlanks = (key, value) => {
+    if (!value) {
+        return (key === 'address' || key === 'name') ? LONG_BLANK : SHORT_BLANK
+    }
+    return value
+}
+
+const _changeToEmptyCell = (key, value) => {
+    if (!value) {
+        return ''
+    }
+    return value
+}
+
+const replaceAllUndefinedValues = (object, replace) => {
+    for (let key in object) {
+        if (typeof object[key] === 'object' && object[key] !== null) {
+            replaceAllUndefinedValues(object[key], replace)
+        } else (
+            object[key] = replace(key, object[key])
+        )
+    }
+    return object
+}
+
+const buildExportWordFile = async ({ task, documentData, locale, timeZone }) => {
+    const { id, ticket } = task
+
+    const { stream } = await buildExportFile({
+        templatePath: `./domains/ticket/templates/ticketDocumentGenerationTemplates/paidWorks/${locale}.docx`,
+        replaces: documentData,
+    })
+
+    return {
+        stream,
+        filename: `paid_works_ticket_${ticket.id}_${formatDateToTimezone(undefined, timeZone, 'DD_MM_YYYY')}.docx`,
+        mimetype: DOCX_FILE_META.mimetype,
+        encoding: DOCX_FILE_META.encoding,
+        meta: {
+            listkey: 'TicketDocumentGenerationTask',
+            id,
+        },
+    }
+}
+
+const generateTicketDocumentOfPaidWorks = async ({ task, baseAttrs, context, locale, ticket, organization }) => {
+    const { iec, psrn, organizationAddress } = await getFinanceInfoDataByLocale(organization, { locale })
+
+    const { format, timeZone: timeZoneFromUser } = task
+
+    const timeZone = normalizeTimeZone(timeZoneFromUser) || DEFAULT_ORGANIZATION_TIMEZONE
+    const printDate = dayjs().tz(timeZone).locale(locale)
+
+    const contact = ticket.contact ? await getByCondition('Contact', {
+        id: ticket.contact,
+        deletedAt: null,
+    }) : null
+
+    const employee = organization.id && ticket.executor
+        ? await getByCondition('OrganizationEmployee', {
+            organization: { id: organization.id },
+            user: { id: ticket.executor },
+            deletedAt: null,
+        })
+        : null
+
+    const invoices = await Invoice.getAll(context, {
+        ticket: { id: ticket.id },
+        deletedAt: null,
+        status_not: INVOICE_STATUS_CANCELED,
+    }, 'id currencyCode accountNumber rows { isMin toPay name count vatPercent } recipient { bankName bankAccount bic }', {
+        sortBy: ['createdAt_ASC'],
+    })
+
+    const currencyCode = get(invoices, '0.currencyCode') || DEFAULT_INVOICE_CURRENCY_CODE
+    let totalSum = Big(0), totalVAT = Big(0)
+
+    const listOfWorks = invoices.reduce((acc, invoice) => {
+        const rows = Array.isArray(invoice.rows) ? invoice.rows : []
+        acc.push(...rows.map((row, index) => {
+            let price
+            if (row.isMin) {
+                price = Big(0)
+            } else {
+                price = !Number.isNaN(parseFloat(row.toPay)) ? Big(row.toPay) : Big(0)
```

Comment:
Why do you use parseFloat instead of Big.js here?

Reply:
If you pass Big(" ") to big.js, there will be an error

Therefore, Number.isNaN(parseFloat(row.toPay) will check that the passed value is not an empty string or a string of spaces